### ISSUE TITLE
Leftover in /system/etc/product/applications and /oem/deletable-app

### ISSUE THREAD

**Issue by randomname1234** [d4rken/sdmaid-public#411]
Device: Sony xperia z4 tablet (SGP712)

Android: 6.0.1 (no custom rom)

root: yes. Rooted with SuperSU v2.74

SD Maid: 4.2.5

Also, I noticed some leftover files of old uninstalled apps in '/system/etc/product/applications' . The only app which is still installed doesn't seem to have this directory listed in AppControl.

Not sure if it is related to the fact that there is a symlink named '/etc' which points to '/system/etc' (probably not as the problem persists with /oem/deletable-app)

> Also note that if a system app gets updated, the installed app is coming from the updated apk file in userspace (e.g. /data/app), but the original apk in /system is still what gives it system app priviledges. SD Maid removes the update on first uninstall and then it should show the app referencing the apk in /system.

I know, but this is not the case. I downloaded and flashed the stock firmware so it was a clean install, no app updated, just the bloatware Sony put there. 

> I'll do some research on this issue, so far couldnt find a device with apps using these locations.

As far as I know, this is the structure:

/system/etc/product/applications - Third party apps (avg antivirus, facebook, navigator, playstation app etc)
/system/vendor/app | Customizations for Chrome (sony homepage) and IMS aurora
/system/vendor/overlay | ALL Sony overlays for some apps
/oem/overlay | same as above
/oem/deletable-app | other leftover here
(I'm not mentioning /system/app or /system/priv-app)

The only problems are with /oem/deletable-app & /system/etc/product/applications.

Directory structure: [http://hastebin.com/emexihunaw.avrasm](http://hastebin.com/emexihunaw.avrasm)

**Comment by d4rken** [comment]
Are those files leftovers from a deliberate actions, i.e. uninstall attempt through AppControl?

Because I'm not in favor of just removing files on /system or /oem because they can be removed as this has more disadvantages than advantages. Extra spaces on these read-only partitions gives us no advantage, but can break stuff like OTA updates or even the ROM.

If these files are left overs from deliberate uninstalls we have to differentiate a few cases:

* Left over because the first uninstall removed the user space update. Which is currently expected behavior, uninstall an updated system app in one go would be #220 
* Left over although SD Maid knew about it, but failed to remove it, that would fall under #410 
* Left over because it was not detected as belonging to an app, thats something we could fix under this ticket

There is also a 4th case where the file wouldn't really be a leftover: The files are there to allow app restoration on factory reset. As AppControl is only responsible for installed apps this would be something you would remove manually through the Explorer.

**Comment by randomname1234** [comment]
> Are those files leftovers from a deliberate actions, i.e. uninstall attempt through AppControl?

My bad, I mean old uninstalled apps through AppControl.

> Left over because the first uninstall removed the user space update. Which is currently expected behavior, uninstall an updated system app in one go would be #220

No, they haven't been updated. The version is the one provided in the (official) firmware I flashed.

> Left over because it was not detected as belonging to an app, thats something we could fix under this ticket

That is more realistic, because (let's say smallbrowser) if I tap on the smallbrowser app in AppControl, it doesn't show me '/system/etc/product/applications/app-smallbrowser-ed' in the list of folders.

> There is also a 4th case where the file wouldn't really be a leftover: The files are there to allow app restoration on factory reset. As AppControl is only responsible for installed apps this would be something you would remove manually through the Explorer.

That is less likely because there are really a few apps in there, compared to the list of bloatware sony put on the tablet (and that consequently get restored after a factory reset)


As for #220 I think it should be:

Delete the app and its data -> delete app, data, updates
Resets the app to its first install -> delete data, updates
Delete data -> delete data

So basically the current 'Resets the app to its first install' would become 'Delete data'
This would be in line with the description provided by the wiki "_Returns the app to it's install state._"

**Comment by d4rken** [comment]
I agree, so "all" that is left now is to somehow figure out how we would check if apps are related to these additional files.

What does AppControl currently show in v4.2.5 for an app such as "Smallbrowser"?

**Comment by randomname1234** [comment]
/data/user/0/com.sonymobile.smallbrowser
/data/app/com.sonymobile.smallbrowser-1
/data/dalvik-cache/profiles/com.sonymobile.smallbrowser

**Comment by randomname1234** [comment]
/data/user/0/com.sonymobile.smallbrowser
/data/app/com.sonymobile.smallbrowser-1
/data/dalvik-cache/profiles/com.sonymobile.smallbrowser

**Comment by d4rken** [comment]
Can you add pathdumps (directory structure) of `/data/user/0/com.sonymobile.smallbrowser` and `/data/app/com.sonymobile.smallbrowser-1`, especially the one in `/data/app` is of interest because normally it should hold the apk for the app.

Generally I'm interested in what AppControl shows for all these types of apps and directory structures of the folders shown by AppControl for these apps.

I think we also did not cover yet whether AppControl actually marks these apps as "system", does it?

**Comment by randomname1234** [comment]
sure, [http://hastebin.com/qaviduwoha.avrasm](http://hastebin.com/qaviduwoha.avrasm)

Smallbrowser is not marked as "system" 

Also, `/data/app/com.sonymobile.smallbrowser-1/base.apk` and `/system/etc/product/applications/app-smallbrowser-ed/app-smallbrowser-ed.apk` are the same.

MD5 hash: 7a35cf3675e45addd1a6eee239c90727

**Comment by d4rken** [comment]
Currently I would think this the 4th case I mentioned.

> >There is also a 4th case where the file wouldn't really be a leftover: The files are there to allow app restoration on factory reset. As AppControl is only responsible for installed apps this would be something you would remove manually through the Explorer.

> That is less likely because there are really a few apps in there, compared to the list of bloatware sony put on the tablet (and that consequently get restored after a factory reset)

I assume you can remove the smallbrowser app without root? I.e. normally, even through system settings > apps? I would also guess that if you then do a factory reset, this app is restored.

**Comment by randomname1234** [comment]
Just checked, apks in `system/etc/applications` and `system/etc/product` are meant to be installed upon fresh setup. Sorry :sweat_smile:

**Comment by d4rken** [comment]
No worries. I wasn't sure either and after investigating it toegether, now we both know.

We could think about marking apps with these properties somehow. For that we would still need to develop some kind of matching. I guess there is no config file that mentioned where what is copied?
